packages/dht/src/dht/PeerManager.ts

```diff
@@ -0,0 +1,339 @@
+/* eslint-disable @typescript-eslint/parameter-properties, class-methods-use-this */
+
+import KBucket from 'k-bucket'
+import { PeerDescriptor } from '../proto/packages/dht/protos/DhtRpc'
+import { DhtPeer } from './DhtPeer'
+import { IPeerManager } from './IPeerManager'
+import { SortedContactList } from './contact/SortedContactList'
+import { RandomContactList } from './contact/RandomContactList'
+import { PeerID, PeerIDKey } from '../helpers/PeerID'
+import { Logger } from '@streamr/utils'
+import { keyFromPeerDescriptor, peerIdFromPeerDescriptor } from '../helpers/peerIdFromPeerDescriptor'
+import { ConnectionManager } from '../connection/ConnectionManager'
+import EventEmitter from 'eventemitter3'
+import { DisconnectionType } from '../transport/ITransport'
+
+const logger = new Logger(module)
+
+export interface PeerManagerConfig {
+    numberOfNodesPerKBucket: number
+    maxNeighborListSize: number
+    getClosestContactsLimit: number
+    ownPeerId: PeerID
+    connectionManager: ConnectionManager
+    nodeName: string
+    createDhtPeer: (peerDescriptor: PeerDescriptor) => DhtPeer
+}
+
+export interface PeerManagerEvents {
```

Comment:
Is the `PeerManager` managing just peers, or also contacts? (if I understood each peer is a contact, but not vice versa)

Reply:
it is managing all Peer-related, also contacts, because contacts are potential Peers

Reply:
Maybe the class should be renamed to `ContactManager` then. The methods also imply that the class handles contacts in general, not just peers:
```
    newContact: (peerDescriptor: PeerDescriptor, closestPeers: PeerDescriptor[]) => void
    contactRemoved: (peerDescriptor: PeerDescriptor, closestPeers: PeerDescriptor[]) => void
    newKbucketContact: (peerDescriptor: PeerDescriptor, closestPeers: PeerDescriptor[]) => void
    kbucketContactRemoved: (peerDescriptor: PeerDescriptor) => void
    newOpenInternetContact: (peerDescriptor: PeerDescriptor, closestPeers: PeerDescriptor[]) => void
    openInternetContactRemoved: (peerDescriptor: PeerDescriptor, closestPeers: PeerDescriptor[]) => void
    newRandomContact: (peerDescriptor: PeerDescriptor, closestPeers: PeerDescriptor[]) => void
    randomContactRemoved: (peerDescriptor: PeerDescriptor, closestPeers: PeerDescriptor[]) => void
```

Reply:
Well, ContactManager sounds like a synonym of AddressBook, which the class is certainly not about. I'd leave the name as-is.

Reply:
As the term `PeerManager`/`ContactManager` is quite abstract, maybe it would makes sense to write a short comment to the class which describes what the core functionality of this class is.

One option is to revisit the naming question when we have the glossary (https://linear.app/streamr/issue/NET-1128/add-to-internal-docs-a-glossary-of-network-level-terms), and therefore also have clear definitions for each name.